java-frontend/src/test/java/org/sonar/java/UCFGJavaVisitorTest.java

```diff
@@ -176,6 +176,49 @@ public void build_concatenate_heterogenous_elements() {
     assertCodeToUCfg("class A {String method(String arg) {return arg+=42;}}", expectedUCFG);
   }
 
+  @Test
+  public void build_primitive_types_to_constants() {
+    Expression.Variable arg1 = UCFGBuilder.variableWithId("arg1");
+    Expression.Variable aux0 = UCFGBuilder.variableWithId("%0");
+    Expression.Variable aux1 = UCFGBuilder.variableWithId("%1");
+    Expression.Variable aux2 = UCFGBuilder.variableWithId("%2");
+    Expression.Variable aux3 = UCFGBuilder.variableWithId("%3");
+    Expression.Variable aux4 = UCFGBuilder.variableWithId("%4");
+    Expression.Variable aux5 = UCFGBuilder.variableWithId("%5");
+    Expression.Variable s = UCFGBuilder.variableWithId("s");
+    Expression.Variable aux6 = UCFGBuilder.variableWithId("%6");
+    Expression.Variable aux7 = UCFGBuilder.variableWithId("%7");
+    Expression.Variable aux8 = UCFGBuilder.variableWithId("%8");
+    Expression.Variable aux9 = UCFGBuilder.variableWithId("%9");
+    UCFG expectedUCFG = UCFGBuilder.createUCFGForMethod("A#method(ZBCS)Ljava/lang/String;").addMethodParam(arg1)
```

Comment:
Add all the parameters (to be consistent), or drop its addition. Adding only one make things blurry.